docs/src/CommutativeAlgebra/groebner_bases.md

```diff
@@ -10,9 +10,155 @@ using Oscar
+F = free_module(R, 2)
+default_ordering(F)
+S, _ = grade(R, [1, 2, 3])
+default_ordering(S)
+```
+
+```@repl oscar
+R, (x, y, z) = PolynomialRing(QQ, ["x", "y", "z"])
+f = 3*z^3+2*x*y+1
+```
+
+## Division With Remainder
+
+Buchberger's algorithm for computing Gröbner bases makes use of reduction steps which rely on multivariate division with remainder.
+For the computation of standard bases, a variant of of the division algorithm due to Mora comes into play. In that context, with notation
+as above, we use the following terminology.
+
+First suppose that $>$ is global and let polynomials $g\in K[x]$ and $f_1, \dots, f_r\in K[x]\setminus \{0\}$ be given.
+In this situation, multivariate division with remainder allows us to compute expressions
+
+$g = q_1f_1+\dots q_rf_r + h, \; h\in K[x], \;\text{ all }\; q_i \in K[x]$
+
+such that:
+
+- If both sides are nonzero, then $\text{LM}_>(g) \ge \text{LM}_>(q_if_i)$.
+- If $h$ is nonzero, then $\text{LM}_>(h)$ is not divisible by any $\text{LM}_>(f_i)$.
+
+Each such expression is called a *standard representation* for $g$ with *quotients* $q_i$ and *remainder* $h$
+(in terms of the $f_i$, with respect to $>$). We say that the remainder $h$ in a standard representation is
+*fully reduced*  if it satisfies the stronger condition below:
+
+- If $h$ is nonzero, then no term of $h$ is divisible by any $\text{LM}_>(f_i)$.
+
+Without restrictions on  $>$, let elements $g\in K[x]_>$ and $f_1, \dots, f_r\in K[x]\setminus \{0\}$ be given.
+In this more general situation, Mora division with remainder allows us to compute expressions
+
+$ug = q_1f_1+\dots q_rf_r + h, \; h\in K[x]_>, \;\text{ all }\; q_i \in K[x]_>$
+
+such that:
+
+- ``u`` is a unit of $ K[x]_>$.
+- If both sides are nonzero, then $\text{LM}_>(g) \ge \text{LM}_>(q_if_i)$.
+- If $h$ is nonzero, then $\text{LM}_>(h)$ is not divisible by any $\text{LM}_>(f_i)$.
+
+Each such expression is called a *weak standard representation* for $g$ with *quotients* $q_i$ and *remainder* $h$ (in terms of the $f_i$, with respect to $>$).
+If $g\in K[x])$, we speak of a *polynomial weak standard representation* if $u$ and the $q_i$ are elements of $K[x].$ While from a theoretical point of view it makes
+still sense to speak of fully reduced remainders, such remainders may not be computable (in finitely many steps).
+
+The OSCAR functions discussed below compute standard representations and polynomial weak standard representations, respectively.
+
+```@julia
+reduce(I::IdealGens, J::IdealGens; ordering::MonomialOrdering = default_ordering(base_ring(J)))
+```
+
+```@julia
+reduce(f::T, F::Vector{T}; ordering::MonomialOrdering = default_ordering(parent(f))) where {T <: MPolyElem}
+```
+
+```@julia
+reduce(F::Vector{T}, G::Vector{T}; ordering::MonomialOrdering = default_ordering(parent(F[1]))) where {T <: MPolyElem}
+```
+
+```@julia
+reduce(F::Vector{T}, G::Vector{T}; ordering::MonomialOrdering = default_ordering(parent(F[1]))) where {T <: MPolyElem}
```

Comment:
Isn't this the same function twice?

Reply:
Sure, just a copy and paste problem. Does not really matter as these are just placeholders for the real functions to come. Nevertheless, I will correct this.